### PR TITLE
adding passport strategy

### PR DESCRIPTION
**Pull request by jonterrylee**
If a strategy is defined, configure passport.  Each strategy must be fully defined.  

Adds the to Req:
 - user: {userId, sub}
 - authInfo: <scope string>

### CODE REVIEW

File: src/server/strategies/bearer.strategy.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	function(token, done) {
+
+		if (!process.env.introspectionUrl) {
+			return done(new Error('Invalid introspection endpoint.'));
+		}
+
+		request
+		.post(process.env.introspectionUrl)
+		.set('content-type', 'application/x-www-form-urlencoded')
+		.send({token: token, client_id: process.env.clientId, client_secret: process.env.clientSecret})
```
Comment (andrewmarcus) [inline src/server/strategies/bearer.strategy.js]: Is it customary to send the token along with the client_id and client_secret used to generate it?  Seems you should keep the secret to yourself?

File: src/server/utils/auth.js [hunk 2]
```diff
@@ -1,226 +0,0 @@
-const _ = require('lodash');
```
Comment (andrewmarcus) [inline src/server/utils/auth.js]: So, does Passport handle the FHIR server end of the transaction, the OAuth server end, or both?

File: src/server/strategies/bearer.strategy.js (hunk 1 above)
Comment (jonterrylee) [reply src/server/strategies/bearer.strategy.js]: yes its required

File: src/server/utils/auth.js (hunk 2 above)
Comment (jonterrylee) [reply src/server/utils/auth.js]: both

**Pull request merged**